README.md

```diff
@@ -64,14 +74,173 @@ If you have a suggestion for a feature, please feel free to file it under "issue
+
+This is great news, as it means that a web-ready version of the library is available!  Just click on the "dist" folder, and find the file you want:  <https://github.com/siciarek/javascript-qrcode/blob/master/dist/qrcode.min.js>
+
+Almost there!  To be consumed inside of Script Lab, we need the file to be served in *RAW* form, ***and*** with correct Content-Type headers.  To do this, use a service like [RawGit](https://rawgit.com) to transform the file for you, and give you a URL you can use directly.
+
+In this case, pasting in the above URL into RawGit produces:
+
+    https://cdn.rawgit.com/siciarek/javascript-qrcode/041ad555/dist/qrcode.min.js
+
+Include this in your Libraries tab, and you're good to go!
+
+
+#### Harder: NPM package is an AMD module (must be loaded with `require.js`), and is *not* available in "dist" form
+
+Suppose you get the same error as in the previous section, but you can't find a dist version of a library that you want.  Let's take [tinygradient](https://github.com/mistic100/tinygradient) as one such example.  The GitHub documentation of tinygradient says that it is "Compatible with Require.js/AMD and NodeJS".  Let's use Require.js to load it.
+
+In your libraries tab, add a reference to `require.js` and to its IntelliSense types (note that `require.js` itself should be loaded from a web-compatible CDN version, not its NPM package):
+
+    https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.3/require.min.js
+    @types/requirejs
+
+Do **NOT** add the names of the actual libraries that you want to load. They will be loaded at runtime instead, see below.
+
+Next, in the Script tab, at the top of the file:
+
+1. Add a line `require.onError = console.error`.  This redirection is vital for seeing useful errors, don't forget it!
+
+2. Add a `require.config` section with a `paths` dictionary, listing out the sources of the JS files, as well as a local alias that you want to give them.  **Importantly**, add the entire path to the JS file, *minus the actual `.js` suffix*.  Thus:
+
+    require.config({
+        paths: {
+            "tinygradient": "https://unpkg.com/tinygradient@0.3.1/tinygradient.min",
+        }
+    });
+
+3. Add the code that you want to use, *inside of a `require` block*:
+
+    require(['tinygradient'], function (tinygradient) {
+        var gradient = tinygradient('red', 'blue');
+        console.log(gradient.rgb(10));
+    });
+
+If you're lucky, it will work right off the bat.  If you're slightly less lucky, you might still get a useful error that will help you fix the issue (which is why the `require.onError` line described in step #1 is so important!)
```

Comment:
Something about this feels weird. Luck has nothing to do with it. 
Its code. there's either a true or a false.